parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -813,7 +841,7 @@ private Type extractField(GroupType candidate, Type targetField) {
       } else {
         Type tempField = extractField(field.asGroupType(), targetField);
         if (tempField != null) {
-          return tempField;
+          return new GroupType(candidate.getRepetition(), candidate.getName(), tempField);
```

Comment:
Is it an individual bug fix?

Reply:
yeah, that's the issue I mentioned in email communication, it become a problem when I started a using a different for the right side file group